TESTS/mbedmicro-rtos-mbed/signals/main.cpp

```diff
@@ -15,49 +15,344 @@
  */
 #include "mbed.h"
 #include "greentea-client/test_env.h"
+#include "unity.h"
+#include "utest.h"
 #include "rtos.h"
 
+using namespace utest::v1;
+
+
 #if defined(MBED_RTOS_SINGLE_THREAD)
   #error [NOT_SUPPORTED] test not supported
 #endif
 
-#define TEST_STACK_SIZE 512
 
-#define SIGNAL_SET_VALUE    0x01
-const int SIGNALS_TO_EMIT = 100;
-const int SIGNAL_HANDLE_DELEY = 25;
+#define TEST_STACK_SIZE   384 /* 512B stack on GCC_ARM compiler cause out of memory on some 16kB RAM boards */
+#define ALL_FLAGS 0x7fffffff
+#define PROHIBITED_FLAG 0x80000000
+#define MAX_FLAG_POS 30
+
+
+#define WAIT1()     sync_sem2.release(); \
+                    sync_sem1.wait()
+
+#define WAIT2()     sync_sem1.release(); \
+                    sync_sem2.wait()
+
+
+Semaphore sync_sem1(0, 1);
+Semaphore sync_sem2(0, 1);
+volatile int32_t ret_val;
+volatile osStatus ret_status;
+
+
+
+void run_set_clear(void)
+{
+    // 1.1
```

Comment:
What does this comment mean?

Reply:
X. - indicates subtest number
X.1, X.2, ... - indicates subtest flow

Reply:
comment added